Fix crash on Android 16 beta caused by missing canImeRenderGesturalNavButtons()

BourbonFather opened the pull request:
Lawnchair crashes immediately on Android 16 beta (e.g., Pixel 8 Pro) due to a missing method in the Android framework:

```
java
InputMethodService.canImeRenderGesturalNavButtons()
```

This method is not present on some builds of Android 16, resulting in a `NoSuchMethodError` at launch.

Fix
Wrapped the method in `isImeRenderingNavButtons()` with a reflection-based check to safely detect and call it only when available. This prevents the launcher from crashing while preserving functionality on supported devices.

Fixes #5535

Type of change
:white_check_mark: Bug fix (non-breaking change which fixes an issue)

BourbonFather committed:
Update RecentsAnimationDeviceState.java

fix android 16 beta crash

adil192 commented:
I couldn't get this to compile since isImeRenderingNavButtons is already defined in the same file:
```java
/home/ahann/Documents/GitHub/lawnchair/quickstep/src/com/android/quickstep/RecentsAnimationDeviceState.java:624: error: method isImeRenderingNavButtons() is already defined in class RecentsAnimationDeviceState
    public boolean isImeRenderingNavButtons() {
                   ^
```

BourbonFather committed:
Update RecentsAnimationDeviceState.java

Fix: remove duplicate method and patch existing isImeRenderingNavButtons()

BourbonFather commented:
Thanks for the heads-up! I removed the duplicate `isImeRenderingNavButtons()` method and updated the existing one to use the reflection-based check as intended.

MrSluffy commented on quickstep/src/com/android/quickstep/RecentsAnimationDeviceState.java:
```diff
@@ -107,6 +107,7 @@ public class RecentsAnimationDeviceState implements DisplayInfoChangeListener, E
     private final RotationTouchHelper mRotationTouchHelper;
     private final TaskStackChangeListener mPipListener;
     // Cache for better performance since it doesn't change at runtime.
+
```
remove this new line

MrSluffy commented on quickstep/src/com/android/quickstep/RecentsAnimationDeviceState.java:
```diff
@@ -609,10 +610,16 @@ public RotationTouchHelper getRotationTouchHelper() {
+            .getDeclaredMethod("canImeRenderGesturalNavButtons");
+        return (Boolean) method.invoke(null);
+    } catch (Exception e) {
+        return false;
```
add this here here `return mCanImeRenderGesturalNavButtons && mMode == NO_BUTTON
                && ((mSystemUiStateFlags & SYSUI_STATE_IME_SHOWING) != 0);`

why? since your reflection is for A16 the old implementation is for below Android version

validcube reviewed (changes requested):
Hello, please check my comment here.

https://github.com/LawnchairLauncher/lawnchair/issues/5535#issuecomment-3361621282

Thanks.
